parsers/ZA.py

```diff
@@ -49,7 +53,7 @@ def fetch_production(
             row_data = row.split(";")[1:]
             if row_data[0] != "":
                 production_csv_data[date] = [
-                    float(item.replace(",", ".")) for item in row.split(";")[1:]
+                    float(item.replace(",", ".")) if float(item.replace(",", ".")) >= 0 else 0 for item in row.split(";")[1:]
```

Comment:
Wouldn't this prevent us from getting the hydro storage values? They have both positive and negative values and both are valid.

It would be better to implement the flooring down on line 124, in this part of the code:
```python
            elif j in production_idcs:
                data["production"][column_mapping[j]] = round(
                    data["production"][column_mapping[j]] + float(prod_data_value),
                    13,
                )
```